rpcs3/Emu/Cell/Modules/cellPngDec.cpp

```diff
@@ -742,13 +840,51 @@ s32 cellPngDecClose(ppu_thread& ppu, PHandle handle, PStream stream)
 s32 cellPngDecReadHeader(PHandle handle, PStream stream, PInfo info)
 {
 	cellPngDec.warning("cellPngDecReadHeader(handle=*0x%x, stream=*0x%x, info=*0x%x)", handle, stream, info);
-	return pngReadHeader(stream, info);
+
+	// Read the header info
+	png_read_info(stream->png_ptr, stream->info_ptr);
+
+	pngSetHeader(stream.get_ptr());
+
+	// Set the pointer to stream info
+	*info = stream->info;
+	return CELL_OK;
 }
 
 s32 cellPngDecExtReadHeader(PHandle handle, PStream stream, PInfo info, PExtInfo extInfo)
 {
 	cellPngDec.warning("cellPngDecExtReadHeader(handle=*0x%x, stream=*0x%x, info=*0x%x, extInfo=*0x%x)", handle, stream, info, extInfo);
-	return pngReadHeader(stream, info, extInfo);
+	// Set the reserved value to 0, if passed to the function. (Should this be arg error if they dont pass?)
+	if (extInfo)
+	{
+		extInfo->reserved = 0;
+	}
+
+	// lets push what we have so far
+	u8* data = static_cast<u8*>(stream->buffer->data.get_ptr()) + stream->buffer->cursor;
+	try 
+	{
+		png_process_data(stream->png_ptr, stream->info_ptr, data, stream->buffer->length);
+	}
+	catch (LibPngCustomException)
```

Comment:
@jarveson and? You are still catching by value. It'd make a copy. Take a look [here](http://stackoverflow.com/questions/7401269/catch-by-reference-when-exception-variable-is-not-defined).